keras/engine/base_layer.py

```diff
@@ -139,26 +138,16 @@ def __init__(self, **kwargs):
             if 'batch_input_shape' in kwargs:
                 batch_input_shape = tuple(kwargs['batch_input_shape'])
             elif 'input_shape' in kwargs:
-                if 'batch_size' in kwargs:
-                    batch_size = kwargs['batch_size']
-                else:
-                    batch_size = None
+                batch_size = kwargs.get('batch_size')
                 batch_input_shape = (
                     batch_size,) + tuple(kwargs['input_shape'])
             self.batch_input_shape = batch_input_shape
 
             # Set dtype.
-            dtype = kwargs.get('dtype')
-            if dtype is None:
-                dtype = kwargs.get('input_dtype')
-            if dtype is None:
-                dtype = K.floatx()
-            self.dtype = dtype
-
-        if 'weights' in kwargs:
-            self._initial_weights = kwargs['weights']
-        else:
-            self._initial_weights = None
+            self.dtype = kwargs.get('dtype') or kwargs.get('input_dtype') \
```

Comment:
Don't use \ for line breaks.

Also: the previous code snippet was a lot more readable than this, please don't change it.

Reply:
Ah, my bad. I've reverted the changes to _that_ snippet, let me know if there are other changes.